olmo/train.py

```diff
@@ -777,6 +811,21 @@ def train_batch(self, batch: Dict[str, Any]) -> Tuple[torch.Tensor, Optional[tor
 
         ce_batch_loss = torch.tensor(0.0, device=self.device)
         z_batch_loss = None if not self.cfg.softmax_auxiliary_loss else torch.tensor(0.0, device=self.device)
+        lb_batch_loss = (
+            None if self.model.config.block_type != BlockType.moe else torch.tensor(0.0, device=self.device)
+        )
+        moe_z_batch_loss = (
+            None if not self.model.config.moe_zloss_weight else torch.tensor(0.0, device=self.device)
+        )
+        # Keep this one on CPU to save memory
+        expert_assignments = (
+            None
+            if (
+                (self.model.config.block_type != BlockType.moe)
+                or (self.model.config.moe_log_expert_assignment is False)
+            )
+            else torch.zeros((self.model.config.n_layers, self.model.config.moe_num_experts))
```

Comment:
This does not put it on CPU. This puts it on the default device.

Reply:
changed to place it on self.device